src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -1835,12 +1835,11 @@ services:
     class: 'PrestaShopBundle\Form\Admin\Sell\Product\DataTransformer\RedirectionTargetTransformer'
     public: true
 
-  form.type.sell.product.event_listener.feature_value_listener:
-    class: 'PrestaShopBundle\Form\Admin\Sell\Product\EventListener\FeatureValueListener'
+  PrestaShopBundle\Form\Admin\Sell\Product\EventListener\FeatureValueListener:
+    public: false
+    autowire: true
     arguments:
-      - '@prestashop.adapter.form.choice_provider.feature_values_choice_provider'
-      - '@PrestaShopBundle\Form\FormCloner'
-    public: true
+      $featureValuesChoiceProvider: '@prestashop.adapter.form.choice_provider.feature_values_choice_provider'
```

Comment:
Why not using autowiring?

Reply:
it depends on interface, so we must define which actual class to inject